sdk/java-sdk/src/main/scala/kalix/javasdk/impl/action/ActionEffectImpl.scala

```diff
@@ -68,6 +72,16 @@ object ActionEffectImpl {
       copy(internalSideEffects = sideEffects)
   }
 
+  final case class IgnoreEffect[T]() extends PrimaryEffect[T] {
+    def isEmpty: Boolean = true
+
+    override def internalSideEffects = Nil
+
+    protected def withSideEffects(sideEffect: Seq[SideEffect]): IgnoreEffect[T] = {
+      throw new IllegalArgumentException("adding side effects to is not allowed.")
+    }
+  }
```

Comment:
Ah I see, PrimaryEffect is invariant, you can do like this to keep it a singleton:

```suggestion
  def IgnoreEffect[T]() = IgnoreEffect.asInstanceOf[PrimaryEffect[T]] 
  case object IgnoreEffect extends PrimaryEffect[Nothing] {
    def isEmpty: Boolean = true

    override def internalSideEffects = Nil

    protected def withSideEffects(sideEffect: Seq[SideEffect]): PrimaryEffect[Nothing] = {
      throw new IllegalArgumentException("adding side effects to is not allowed.")
    }
  }
```

Reply:
Or just cast it in `ignore()` since that is the only place referencing it needing to assign a type param

Reply:
I prefer your first solution. I think is more expressive.